src/annotator/anchoring/test/xpath-util-test.js

```diff
@@ -0,0 +1,320 @@
+import $ from 'jquery';
+
+import {
+  findChild,
+  getTextNodes,
+  getLastTextNodeUpTo,
+  getFirstTextNodeNotBefore,
+  simpleXPathPure,
+  simpleXPathJQuery,
+} from '../xpath-util';
+
+describe('annotator/anchoring/xpath-util', () => {
+  describe('#findChild', () => {
+    let container;
+
+    beforeEach(() => {
+      container = document.createElement('div');
+      document.body.appendChild(container);
+    });
+
+    afterEach(() => {
+      container.remove();
+    });
+
+    it('throws an error when the parent has no children', () => {
+      assert.throws(() => {
+        findChild(container, 'fake', 0);
+      }, 'XPath error: node has no children!');
+    });
+
+    it('throws an error when the desired child is not found', () => {
+      container.innerHTML = `<div>text</div>`;
+      assert.throws(() => {
+        findChild(container, 'fake', 0);
+      }, 'XPath error: wanted child not found.');
+    });
+
+    describe('with children', () => {
+      beforeEach(() => {
+        container = document.createElement('div');
```

Comment:
Is the override of `container` and the `afterEach` needed here? It seems to me like you could just re-use the container created in the containing block.

Reply:
for sure!